src/core/server/saved_objects/saved_objects_type_registry.test.ts

```diff
@@ -25,25 +25,66 @@ describe('SavedObjectTypeRegistry', () => {
     registry = new SavedObjectTypeRegistry();
   });
 
-  it('allows to register types', () => {
-    registry.registerType(createType({ name: 'typeA' }));
-    registry.registerType(createType({ name: 'typeB' }));
-    registry.registerType(createType({ name: 'typeC' }));
-
-    expect(
-      registry
-        .getAllTypes()
-        .map((type) => type.name)
-        .sort()
-    ).toEqual(['typeA', 'typeB', 'typeC']);
-  });
+  describe('#registerType', () => {
+    it('allows to register types', () => {
+      registry.registerType(createType({ name: 'typeA' }));
+      registry.registerType(createType({ name: 'typeB' }));
+      registry.registerType(createType({ name: 'typeC' }));
+
+      expect(
+        registry
+          .getAllTypes()
+          .map((type) => type.name)
+          .sort()
+      ).toEqual(['typeA', 'typeB', 'typeC']);
+    });
 
-  it('throws when trying to register the same type twice', () => {
-    registry.registerType(createType({ name: 'typeA' }));
-    registry.registerType(createType({ name: 'typeB' }));
-    expect(() => {
+    it('throws when trying to register the same type twice', () => {
       registry.registerType(createType({ name: 'typeA' }));
-    }).toThrowErrorMatchingInlineSnapshot(`"Type 'typeA' is already registered"`);
+      registry.registerType(createType({ name: 'typeB' }));
+      expect(() => {
+        registry.registerType(createType({ name: 'typeA' }));
+      }).toThrowErrorMatchingInlineSnapshot(`"Type 'typeA' is already registered"`);
+    });
+
+    it('throws when `management.onExport` is specified but `management.importableAndExportable` is not or false', () => {
```

Comment:
nit
```suggestion
    it('throws when `management.onExport` is specified but `management.importableAndExportable` is undefined or false', () => {
```

Reply:
We should probably express with the type system as well using a union of which combinations of options are valid